### PR TITLE
MEX-46, MEX-47, updates and deletes

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

- Implement update and delete
- Test the update and delete functionality

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-s3/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: destination/destination.go [hunk 1]
```diff
@@ -17,24 +17,33 @@ package destination
...
 	metadataTable  = "table"
 	metadataAction = "action"
 
 	// action names.
 	actionInsert = "insert"
+	actionUpdate = "update"
+	actionDelete = "delete"
+)
+
+var (
```
Comment (maksenius) [inline destination/destination.go:40]: maybe lets create file error.go and move errors to this file

File: destination/destination.go [hunk 2]
```diff
@@ -136,19 +242,20 @@ func (d *Destination) extractColumnsAndValues(payload sdk.StructuredData) ([]str
...
 	}
 
 	return structuredData, nil
 }
 
-// getTableName returns either the records metadata value for table or the default configured value for table.
+// getTableName returns either the records metadata value for table
+// or the default configured value for table.
 func (d *Destination) getTableName(metadata map[string]string) string {
 	tableName, ok := metadata[metadataTable]
```
Comment (maksenius) [inline destination/destination.go:260]: is it good than metadata has more priority then config?

Comment (alyakimenko) [reply destination/destination.go:260]: In this way, we can handle multiple tables within a single connector. In the config, we set the default table, but if a `table` field was passed in the metadata, we use it instead

File: destination/destination.go (hunk 1 above)
Comment (alyakimenko) [reply destination/destination.go]: Moved the errors to a separate file

File: destination/destination.go [hunk 3]
```diff
@@ -117,13 +125,101 @@ func (d *Destination) insert(ctx context.Context, record sdk.Record) error {
...
+		return fmt.Errorf("failed to get payload: %w", err)
+	}
+
+	// if payload is empty we don't need to insert anything
+	if payload == nil {
+		return ErrEmptyPayload
+	}
+
+	// remove key from the payload, we will use the key inside a WHERE clause.
+	delete(payload, keyColumnName)
```
[the remark this answers is not in the document]
Comment (alyakimenko) [reply destination/destination.go:161]: 1. It's a go's built-in function, and it doesn't return an error

[the remark this answers is not in the document]
Comment (senseyman) [reply destination/destination.go:161]: my bad, I thought it was call for `func (d *Destination) delete`

**Pull request merged**